packages/serverpod/lib/src/database/database_query.dart

```diff
@@ -26,11 +26,15 @@ class SelectQueryBuilder {
   bool _joinOneLevelManyRelationWhereExpressions = false;
   bool _wrapWhereInNot = false;
   TableRelation? _countTableRelation;
+  final bool _viewTable;
 
   /// Creates a new [SelectQueryBuilder].
   /// Throws an [ArgumentError] if the table has no columns.
-  SelectQueryBuilder({required Table table})
-      : _table = table,
+  SelectQueryBuilder({
+    required table,
+    bool viewTable = false,
```

Comment:
Suggestion: Deduce this from the table.

There is a risk that this gets out of sync with the table that is passed in, and we make assumptions around them being in sync in the code.

Better to tie them close together and retrieve this property from the table passed in.